core/plugins/org.polarsys.capella.core.platform.eclipse.ui.trace/src/org/polarsys/capella/core/platform/eclipse/capella/ui/trace/extension/TraceExtensionManager.java

```diff
@@ -21,24 +22,35 @@
 import org.eclipse.core.runtime.IConfigurationElement;
 import org.eclipse.core.runtime.IExtensionRegistry;
 import org.eclipse.core.runtime.Platform;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.ResourceSet;
 import org.polarsys.capella.common.data.modellingcore.AbstractTrace;
 import org.polarsys.capella.core.data.capellacore.Trace;
+import org.polarsys.kitalpha.ad.services.manager.ViewpointManager;
+import org.polarsys.kitalpha.ad.services.manager.ViewpointManager.Listener2;
+import org.polarsys.kitalpha.resourcereuse.model.Resource;
 
 public class TraceExtensionManager {
 
   public static TraceExtensionManager eINSTANCE = new TraceExtensionManager();
-  static String EXTENSION_POINT_ID = "org.polarsys.capella.core.platform.eclipse.capella.ui.trace.extension";
-  static String EXTENSION_POINT_CLASS = "traceExtension";
-
+  String EXTENSION_POINT_ID = "org.polarsys.capella.core.platform.eclipse.capella.ui.trace.extension";
+  String EXTENSION_POINT_CLASS = "traceExtension";
+  private HashMap<ITraceExtension, Set<ResourceSet>> activatedExtensions ;
```

Comment:
Storing the resourceset will lead to memory leaks. as when session is closed, it will be remain here.
Does ViewpointManager.getInstance(rSet).isActive(anExtension.getViewpointID()) in your inActive method is not enough?

Reply:
Yup you're right, removed the cache and now directly use vpManager.isActive